### ISSUE TITLE
Warnings during the build

### ISSUE THREAD

**Issue by atr-sysadmins** [nrk/phpiredis#36]
Building from source, we get:

```
/bin/bash /var/tmp/php-iredis-1.0.0/libtool --mode=compile cc  -I. -I/var/tmp/php-iredis-1.0.0 -DPHP_ATOM_INC -I/var/tmp/php-iredis-1.0.0/include -I/var/tmp/php-iredis-1.0.0/main -I/var/tmp/php-iredis-1.0.0 -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /var/tmp/php-iredis-1.0.0/phpiredis.c -o phpiredis.lo 
libtool: compile:  cc -I. -I/var/tmp/php-iredis-1.0.0 -DPHP_ATOM_INC -I/var/tmp/php-iredis-1.0.0/include -I/var/tmp/php-iredis-1.0.0/main -I/var/tmp/php-iredis-1.0.0 -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /var/tmp/php-iredis-1.0.0/phpiredis.c  -fPIC -DPIC -o .libs/phpiredis.o
/var/tmp/php-iredis-1.0.0/phpiredis.c: In function 'zif_phpiredis_multi_command':
/var/tmp/php-iredis-1.0.0/phpiredis.c:224:9: warning: passing argument 2 of 'redisGetReply' from incompatible pointer type [enabled by default]
In file included from /var/tmp/php-iredis-1.0.0/phpiredis.c:1:0:
/usr/include/hiredis/hiredis.h:194:5: note: expected 'void **' but argument is of type 'struct redisReply **'
/var/tmp/php-iredis-1.0.0/phpiredis.c: In function 'zif_phpiredis_multi_command_bs':
/var/tmp/php-iredis-1.0.0/phpiredis.c:294:9: warning: passing argument 3 of 'redisAppendCommandArgv' from incompatible pointer type [enabled by default]
In file included from /var/tmp/php-iredis-1.0.0/phpiredis.c:1:0:
/usr/include/hiredis/hiredis.h:205:5: note: expected 'const char **' but argument is of type 'char **'
/var/tmp/php-iredis-1.0.0/phpiredis.c:313:9: warning: passing argument 2 of 'redisGetReply' from incompatible pointer type [enabled by default]
In file included from /var/tmp/php-iredis-1.0.0/phpiredis.c:1:0:
/usr/include/hiredis/hiredis.h:194:5: note: expected 'void **' but argument is of type 'struct redisReply **'
/var/tmp/php-iredis-1.0.0/phpiredis.c: In function 'zif_phpiredis_command_bs':
/var/tmp/php-iredis-1.0.0/phpiredis.c:420:5: warning: passing argument 3 of 'redisAppendCommandArgv' from incompatible pointer type [enabled by default]
In file included from /var/tmp/php-iredis-1.0.0/phpiredis.c:1:0:
/usr/include/hiredis/hiredis.h:205:5: note: expected 'const char **' but argument is of type 'char **'
/var/tmp/php-iredis-1.0.0/phpiredis.c:428:5: warning: passing argument 2 of 'redisGetReply' from incompatible pointer type [enabled by default]
In file included from /var/tmp/php-iredis-1.0.0/phpiredis.c:1:0:
/usr/include/hiredis/hiredis.h:194:5: note: expected 'void **' but argument is of type 'struct redisReply **'
/var/tmp/php-iredis-1.0.0/phpiredis.c: In function 'zif_phpiredis_format_command':
/var/tmp/php-iredis-1.0.0/phpiredis.c:487:5: warning: passing argument 3 of 'redisFormatCommandArgv' from incompatible pointer type [enabled by default]
In file included from /var/tmp/php-iredis-1.0.0/phpiredis.c:1:0:
/usr/include/hiredis/hiredis.h:163:5: note: expected 'const char **' but argument is of type 'char **'
/bin/bash /var/tmp/php-iredis-1.0.0/libtool --mode=link cc -DPHP_ATOM_INC -I/var/tmp/php-iredis-1.0.0/include -I/var/tmp/php-iredis-1.0.0/main -I/var/tmp/php-iredis-1.0.0 -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -o phpiredis.la -export-dynamic -avoid-version -prefer-pic -module -rpath /var/tmp/php-iredis-1.0.0/modules  phpiredis.lo -lhiredis
libtool: link: cc -shared  -fPIC -DPIC  .libs/phpiredis.o   -lhiredis  -O2   -Wl,-soname -Wl,phpiredis.so -o .libs/phpiredis.so
libtool: link: ( cd ".libs" && rm -f "phpiredis.la" && ln -s "../phpiredis.la" "phpiredis.la" )
/bin/bash /var/tmp/php-iredis-1.0.0/libtool --mode=install cp ./phpiredis.la /var/tmp/php-iredis-1.0.0/modules
libtool: install: cp ./.libs/phpiredis.so /var/tmp/php-iredis-1.0.0/modules/phpiredis.so
libtool: install: cp ./.libs/phpiredis.lai /var/tmp/php-iredis-1.0.0/modules/phpiredis.la
libtool: finish: PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/sbin" ldconfig -n /var/tmp/php-iredis-1.0.0/modules
```

hiredis is latest master. Are these warning harmless?

**Comment by seppo0010** [comment]
They look harmless. For example, expecting void**, but sending a struct redisReply ** is OK. We could add an explicit cast or a temporary variable to avoid the warning, but it is harmless as is.

**Comment by nrk** [comment]
Warnings should have gone with the `php7` branch (despite the name it also fixes some stuff for PHP 5.x builds) which should be merged into master soon, see #50 for details.
